staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/conversion/converter.go

```diff
@@ -20,27 +20,66 @@ import (
 	"fmt"
 
 	"k8s.io/apiextensions-apiserver/pkg/apis/apiextensions"
+	apiextensionsfeatures "k8s.io/apiextensions-apiserver/pkg/features"
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/apimachinery/pkg/runtime/schema"
+	utilfeature "k8s.io/apiserver/pkg/util/feature"
 )
 
+type CRDConverterFactory struct {
+	// webhookConverterFactory is the factory for webhook converters.
+	// This field should not be used if CustomResourceWebhookConversion feature is disabled.
+	webhookConverterFactory *webhookConverterFactory
+}
+
+func NewCRDConverterFactory() (*CRDConverterFactory, error) {
+	converterFactory := &CRDConverterFactory{}
+	if utilfeature.DefaultFeatureGate.Enabled(apiextensionsfeatures.CustomResourceWebhookConversion) {
```

Comment:
I would prefer to set this up even for disable feature gate, but not call the actual conversion depending on the gate. That's easier to reason about.

Reply:
I thought about it. The thing is if the creating of this factory fails for any reason, with turning feature gate off, we can bypass it but if we don't feature gate it, the conversions will be fail even if we don't use webhooks.

Reply:
When can it fail?

Reply:
right now `newWebhookConverterFactory` can fail if one of it's sub-calls fails. We can go into the details of those subcalls (`NewClientManager` and `NewDefaultAuthenticationInfoResolver`) but the fact that they can fail and even if we resolve failure for current code, it can change in future, make me believe that we should protect ourself from that by feature gate.